### ISSUE TITLE
"Error: Invalid method arguments" when using promises

### ISSUE THREAD

**Issue by samvv** [neovim/node-client#26]
When cloning the current source, running `npm i`, and then `make`, I get the following error from mocha:

```
(node:91429) UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 1): Error: Invalid method arguments
(node:91429) DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```

Source of the problem seems to be `this` not getting passed properly when using Buffer, Window, or Tabpage.

I'm experiencing the same problem in my source. This doesn't work:

```ts
const pos = await win.getPosition()
```
This almost works, but throws a different error:

```ts
const pos = await win.getPosition(win)
```

**Comment by samvv** [comment]
I'm working on a pull request which can solve the issue.

**Comment by billyvg** [comment]
Closed #28
